packages/perseus/src/widgets/radio/choice.jsx

```diff
@@ -9,192 +9,115 @@ import {Popover, PopoverContent} from "@khanacademy/wonder-blocks-popover";
+    // an object with two keys: `checked` and `crossedOut`. Each contains a
+    // boolean value specifying the new checked and crossed-out value of
+    // this choice.
+    onChange: (newValues: {checked: boolean, crossedOut: boolean}) => void,
+|};
 
-        // Option status is exclued for SAT
-        if (this.props.apiOptions.satStyling) {
-            return null;
-        }
+type WithForwardRef = {|forwardedRef: React.Ref<"button">|};
 
-        // Option status is shown only in review mode, or for incorrectly
-        // answered items.
-        if (!reviewMode && !previouslyAnswered) {
-            return null;
-        }
+type ChoicePropsWithForwardRef = {|
+    ...ChoiceProps,
+    ...WithForwardRef,
+|};
 
-        return (
-            <OptionStatus
-                checked={checked}
-                correct={correct}
-                crossedOut={crossedOut}
-                previouslyAnswered={previouslyAnswered}
-            />
-        );
-    };
-
-    renderChoiceIcon: (args: {|
-        isFocused: boolean,
-        isHovered: boolean,
-        isPressed: boolean,
-    |}) => React.Node = ({isFocused, isHovered, isPressed}) => {
-        const {radioStyleVersion, primaryProductColor} =
-            this.props.apiOptions.styling;
-        const finalStyles =
-            typeof radioStyleVersion === "undefined"
-                ? false
-                : radioStyleVersion === "final";
-
-        if (!finalStyles && !this.props.apiOptions.satStyling) {
-            return null;
+function Choice(props: ChoicePropsWithForwardRef): React.Node {
+    const {
+        disabled,
+        checked,
+        content,
+        crossedOut,
+        showCorrectness,
+        multipleSelect,
+        onChange,
+        reviewMode,
+        correct,
+        apiOptions,
+        previouslyAnswered,
+        pos,
+        showRationale,
+        rationale,
+        forwardedRef,
+    } = props;
+    const [isInputFocused, setIsInputFocused] = useState(false);
+
+    useEffect(() => {
+        if (isInputFocused && disabled) {
+            setIsInputFocused(false);
         }
-
-        return (
-            <ChoiceIcon
-                pos={this.props.pos}
-                correct={this.props.correct}
-                crossedOut={this.props.crossedOut}
-                focused={isFocused}
-                hovered={isHovered}
-                pressed={isPressed}
-                checked={this.props.checked}
-                multipleSelect={this.props.multipleSelect}
-                showCorrectness={this.props.showCorrectness}
-                reviewMode={this.props.reviewMode}
-                product={this.props.apiOptions.satStyling ? "sat" : "library"}
-                primaryProductColor={primaryProductColor}
-                previouslyAnswered={this.props.previouslyAnswered}
-            />
-        );
-    };
+    }, [disabled, isInputFocused, setIsInputFocused]);
```

Comment:
Thanks for adding the dep list.  We have a lint rule for this in webapp that we should add to Perseus.  I can add that in a separate PR.